pyccl/core.py

```diff
@@ -303,9 +308,10 @@ def read_yaml(cls, filename):
         return cls(**inits)
 
     def _build_config(
-            self, transfer_function, matter_power_spectrum,
```

Comment:
Isn't it a bit dangerous that an internal facing function takes default values and optional arguments? Shouldn't the internal code always know what to passin? In addition, the docstring didn't say what None stands for.

Reply:
The internal function is only ever called by us so I’m not worried about the defaults. 

As for a doc string, I would think the name of the function is sufficient.

Reply:
Sure. It was just a random thought that popped up when I saw this. Now I realized my concept flow was 'optional arg' => 'syntax sugar' => 'public api', then I blindly inverted it.